beacon-chain/db/state.go

```diff
@@ -44,7 +47,7 @@ func (db *BeaconDB) InitializeState(ctx context.Context, genesisTime uint64, dep
 	blockRoot, _ := hashutil.HashBeaconBlock(genesisBlock)
 	// #nosec G104
 	blockEnc, _ := proto.Marshal(genesisBlock)
-	zeroBinary := encodeSlotNumber(0)
+	zeroBinary := encodeSlotNumberRoot(params.BeaconConfig().GenesisSlot, blockRoot)
```

Comment:
why zero contains encode slot number root of (params.BeaconConfig().GenesisSlot, blockRoot)?

Reply:
Good q @shayzluf , we can have multiple blocks saved at a single slot, so this way, we can prefix search for them in the DB. I will add more documentation to explain this.